core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -741,7 +741,11 @@ public String toString() {
         QUERY_SHARD_EXCEPTION(org.elasticsearch.index.query.QueryShardException.class,
                 org.elasticsearch.index.query.QueryShardException::new, 141),
         NO_LONGER_PRIMARY_SHARD_EXCEPTION(ShardStateAction.NoLongerPrimaryShardException.class,
-                ShardStateAction.NoLongerPrimaryShardException::new, 142);
+            ShardStateAction.NoLongerPrimaryShardException::new, 142),
```

Comment:
That’s an interesting idea and I agree more tailor made exceptions are a shame. I’ll look at that approach…


> On 28 Mar 2016, at 20:48, Simon Willnauer <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/ElasticsearchException.java:
> 
> > @@ -741,7 +741,11 @@ public String toString() {
> >          QUERY_SHARD_EXCEPTION(org.elasticsearch.index.query.QueryShardException.class,
> >                  org.elasticsearch.index.query.QueryShardException::new, 141),
> >          NO_LONGER_PRIMARY_SHARD_EXCEPTION(ShardStateAction.NoLongerPrimaryShardException.class,
> > -                ShardStateAction.NoLongerPrimaryShardException::new, 142);
> > +            ShardStateAction.NoLongerPrimaryShardException::new, 142),
> 
> I think the last thing we need is another exception. We should get rid of the instanceof checks in order to do retries and use ElasticsearchException#addHeader I think we can just use IllegalIndexShardStateException instead?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>